### DISCUSSION TITLE
[New Feature] Helicopter rotor blades animation

### DISCUSSION THREAD

**Discussion by Lupus-Mechanicus** [Vinifera-Developers/Vinifera#1527] [Feature Requests]
### **Description:** 

Unlike in **Red Alert 2** [and **Yuri's Revenge** of course], currently the **Tiberian Sun** engine doesn't allow you to introduce animations for voxels or SHPs using the usual HVA files. This results in weird rotor-less helicopters and VTOLs, and it forces modelers in the best case to either make static rotor blades which look very odd, or in the worse case, to ditch them out entirely.

While it was experimented on before, using both the **TurretSpin** method or the **HVA** one, the results were as expected. They don't really work.

**Screenshots (if appropriate):**
Here you can see as an example, the _Harpy [NOD]_, on a clear green background, missing the rotor blades.
![image](https://user-images.githubusercontent.com/63616666/117195603-2173f000-adee-11eb-95cc-acc526457ebf.png)

An animated example worth looking at is the OpenRA iteration, one version with short rotor blades, and one with longer rotor blades, though as you can clearly see, the shadow for the rotor blades is missing.
![image](https://user-images.githubusercontent.com/11763228/89176050-dbd08100-d556-11ea-933d-bfe0745ae9fb.gif)

**Comment by CCHyper** [comment]
Thanks for mentioning that OpenRA has added support for this, it might help with some of the tilt calculations.

Does OpenRA support aircraft with rotors that pitch and roll?

**Comment by Lupus-Mechanicus** [comment]
From what I've gathered so far, OpenRA supports aircraft/helicopters that pitch and roll [due to [PR#18231](https://github.com/OpenRA/OpenRA/pull/18231)], but I honestly haven't got a clue about aircraft/helicopters with rotors that pitch and roll. :man_facepalming:

pchote said this a while back on this subject [on their side]:
> I explicitly accounted for this in #18231 and chose pitch/roll angles that are large enough to be noticeable, but small enough to still look fine with the sprite rotor.
